### PR TITLE
feat: add deployment UI

### PR DESCRIPTION
**Pull request by JayCeeJr**
https://github.com/go-vela/community/issues/111

Added Deployment form to UI.

### CODE REVIEW

File: src/elm/Nav.elm [hunk 1]
```diff
@@ -298,6 +298,7 @@ viewRepoTabs rm org repo currentPage =
             , Tab "Secrets" currentPage (Pages.RepoSecrets "native" org repo Nothing Nothing) False
             , Tab "Audit" currentPage (Pages.Hooks org repo rm.hooks.maybePage rm.hooks.maybePerPage) isAlerting
             , Tab "Settings" currentPage (Pages.RepoSettings org repo) False
+            , Tab "Deployments" currentPage (Pages.AddDeployment org repo) False
```
[the remark this answers is not in the document]
Comment (davidvader) [reply src/elm/Nav.elm:301]: Notice for secrets, that we have a Secrets tab that displays the resource, then theres a button to get to the Add page, which isnt a tab. 
Screen Shot 2021-07-13 at 9 59 54 AM

**Review by davidvader — CHANGES_REQUESTED** [review]
overall well done, very nice feature addition! 
I left some feedback in the comments, let me know if you have questions.

File: src/elm/Crumbs.elm [hunk 2]
```diff
@@ -177,6 +177,16 @@ toPath page =
...
 
+                Pages.AddDeployment org repo ->
+                    let
+                        orgPage =
+                            ( org, Nothing )
+
+                        currentRepo =
+                            ( repo, Just <| Pages.RepositoryBuilds org repo Nothing Nothing Nothing )
+                    in
+                    [ overviewPage, orgPage, currentRepo, ( "Deployment", Nothing ) ]
```
[the remark this answers is not in the document]
Comment (davidvader) [reply src/elm/Crumbs.elm:188]: Screen Shot 2021-07-13 at 10 05 01 AM

File: src/elm/Nav.elm (hunk 1 above)
[the remark this answers is not in the document]
Comment (davidvader) [reply src/elm/Nav.elm:301]: we might be better off replicating the experience we get with the Audit tab, but for deployments.

for example, the deployments tab would would render a table of deployments returned by `vela get deployment`, and on it there is a button `Add Deployment` that would take you to a separate page for adding a deployment.
see Audit (hooks) for a good example

[the remark this answers is not in the document]
Comment (JayCeeJr) [reply src/elm/Nav.elm:301]: Do we want to add the list of deployments to this PR? 

I agree that having a list and having a button to add one would be a good experience. However until that list is ready it doesn't make much sense to have a blank page with a button for adding deployments. I think we should add another story along with a button to repeat or promote a deployment that could fill out the form

File: src/scss/_main.scss [hunk 3]
```diff
@@ -1364,7 +1364,8 @@ nav {
   font-size: 1.2rem;
 }
```
[the remark this answers is not in the document]
Comment (JayCeeJr) [reply src/scss/_main.scss:1366]: I added some margin and padding around these elements, I also changed the help link to open in a new tab.
While I was in there I fixed the help link in secrets and made that open in a new tab as well.

File: src/elm/Help/Commands.elm [hunk 4]
```diff
@@ -724,6 +754,9 @@ resourceLoaded args =
         Pages.RepositoryBuilds _ _ _ _ _ ->
             args.builds.success
 
+        Pages.RepositoryDeployments _ _ _ _ ->
+            args.builds.success
```
Comment (JayCeeJr) [inline src/elm/Help/Commands.elm:758]: I'm not sure what this is doing.

File: src/elm/Help/Commands.elm [hunk 5]
```diff
@@ -799,6 +838,9 @@ resourceLoading args =
         Pages.RepositoryBuilds _ _ _ _ _ ->
             args.builds.loading
 
+        Pages.RepositoryDeployments _ _ _ _ ->
+            args.builds.loading
```
Comment (JayCeeJr) [inline src/elm/Help/Commands.elm:842]: Same here, what should this be?

Comment (davidvader) [reply src/elm/Help/Commands.elm:842]: sorry for the late reply, i missed this comment!
see my explanation in the org page review:
https://github.com/go-vela/ui/pull/424#discussion_r680065056

**Review by wass3r — APPROVED** [review]

**Pull request merged**